### ISSUE TITLE
TalkBack incorrectly reading links in nested spans

### ISSUE THREAD

**Issue by nicola-d-bbc** [bbc/simorgh#9561]
**Describe the bug**
When using TalkBack on Android, links with role text and nested spans aren't read as one item as intended to be. Content is also repeated in additional blocks/gestures.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to https://www.bbc.com/portuguese/podcasts/p07r3r3t
2. Navigate to the 'O podcast está disponível em' header on TalkBack
3. Swipe to the list on TalkBack
4. TalkBack should read 'Spotify, Que Historia, externo, link'
5. Swipe through the links and TalkBack should read an additional 'Spotify', and the text highlighted
6. Swipe again and TalkBack will read out 'Que Historia' as well as highlight a separate 'block' on the screen.

**Expected behaviour**
TakBack should only read out 'Spotify, Que Historia, externo, link' before moving on to the next link.

**Screenshots**
![image](https://user-images.githubusercontent.com/52782713/137736876-c9d8c6cf-06ac-4733-8f6f-ab9995b9ea06.png)

**Smartphone (please complete the following information):**

- Device: Samsung S10
- OS: Android 11
- Browser: Chrome
- Version: 94.0.4606.85

**Testing notes**

- [x] This bug fix is expected to need manual testing.

**Checklist**
- [x] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

**Comment by DarioR01** [comment]
Each list item has a link with an aria-label consisting of the link name and podcast title. Further down we have 2 span that serves the exact same purpose, of which the second one is visually hidden.
Screen readers will first read the aria-label on the 'a' link for then reading each span (including the visually hidden one). This causes the behaviour experienced in this bug.

However this onlt happens with TalkBack

<img width="989" alt="Screenshot 2021-10-20 at 09 30 57" src="https://user-images.githubusercontent.com/90621252/138058126-7bff2b7e-84eb-4c6a-9d44-54ceb93b9e1c.png">

To fix this bug we need to restructure the PodcastExternalLink component in Simorgh. This can be found [here](https://github.com/bbc/simorgh/blob/latest/src/app/containers/PodcastExternalLinks/index.jsx) . 

<img width="658" alt="Screenshot 2021-10-20 at 09 51 31" src="https://user-images.githubusercontent.com/90621252/138061006-8a8f6e09-4448-4673-8ced-9c2f0dbc4928.png">

Might be worth waiting for #9522 to be merged

**Comment by pvaliani** [comment]
it might be worth checking if https://github.com/bbc/simorgh/pull/9566 has any effect on this ticket? The PR has been merged now so should be able to check on latest

**Comment by nicola-d-bbc** [comment]
This can be fixed by adding aria-labelledby to `<a>` tags with nested spans, and adding a unique ID to spans with `role="text"` as this allows these to be treated as one gesture. Currently nested spans are treated as individual gestures which is a bug with TalkBack. This fix can be removed once the bug has been fixed on their end.
